File: backend/WarehouseInventory.Api/Contracts/Products/ProductListItemDto.cs
namespace WarehouseInventory.Api.Contracts.Products;

public record ProductListItemDto(
    Guid Id,
    string Sku,
    string? Barcode,
    string Name,
    string Category,
    string Location,
    decimal Price,
    int CurrentStock,
    int ReorderThreshold,
    bool IsLowStock,
    Guid Version
);

File: frontend/src/app/data/dto/product-list-item.dto.ts
export interface ProductListItemDto {
  id: string;
  sku: string;
  barcode: string | null;
  name: string;
  category: string;
  location: string;
  price: number;
  currentStock: number;
  reorderThreshold: number;
  isLowStock: boolean;
  version: string;
}
